qutip/solver/sode/rouchon.py

```diff
@@ -132,5 +160,13 @@ def options(self):
     def options(self, new_options):
         Integrator.options.fset(self, new_options)
 
+    def reset(self, hard=False):
+        if self._is_set:
+            state = self.get_state()
+        if hard:
+            raise NotImplementedError
```

Comment:
```suggestion
            raise NotImplementedError("hard reset not implemented")
```

Reply:
How come this has a `NotImplementedError` but some other `reset` implementations just ignore the `hard` option?